pwa/src/components/translations/tableNamesTable.tsx

```diff
@@ -1,168 +1,110 @@
 import * as React from "react";
-import {
-  Table,
-  Card,
-  Spinner,
-  Alert,
-  Modal,
-} from "@conductionnl/nl-design-system/lib";
-import { isLoggedIn } from "../../services/auth";
 import { Link } from "gatsby";
-import FlashMessage from "react-flash-message";
+import { Table, Card, Spinner, Alert } from "@conductionnl/nl-design-system/lib";
 import APIService from "../../apiService/apiService";
 import APIContext from "../../apiService/apiContext";
 
 export default function TableNamesTable() {
   const [documentation, setDocumentation] = React.useState<string>(null);
   const [tableNames, setTableNames] = React.useState<Array<any>>(null);
-  const [context, setContext] = React.useState(null);
   const [showSpinner, setShowSpinner] = React.useState<boolean>(false);
   const [alert, setAlert] = React.useState(null);
   const API: APIService = React.useContext(APIContext);
-
+  
   React.useEffect(() => {
-    if (typeof window !== "undefined" && context === null) {
-      setContext({
-        adminUrl: process.env.GATSBY_ADMIN_URL,
-      });
-    } else if (isLoggedIn()) {
-      getTranslations(context);
-    }
-  }, [context]);
+    getTableNames()
+  }, [API]);
 
-  const getTranslations = (context) => {
+  const getTableNames = () => {
     setShowSpinner(true);
-    fetch(`${context.adminUrl}/table_names`, {
-      headers: {
-        "Content-Type": "application/json",
-        Authorization: "Bearer " + sessionStorage.getItem("jwt"),
-      },
-    })
-      .then((response) => response.json())
-      .then((data) => {
-        // convert array to array objects
-        const convertedArray = data["results"].map((value) => ({
-          name: value,
-        }));
-        setShowSpinner(false);
-        setTableNames(convertedArray);
+    API.Translation.getTableNames()
+      .then((res) => { 
+        let names = [];
+        let results = res.data.results;
+        for (let i = 0; i < results.length; i++) {
+          names.push({name: results[i]})
+        }
+        setTableNames(names); 
```

Comment:
Generally speaking, when you're trying to map an array into new data, you could best use the `.map()` method (instead of a `for` loop). Like so,

```
const names = res.data.results.map((name) => { 
  return { name: name }
});
```

And since `name` is also the key, we can even simplify `name: name` to `{ ... {name} }`, like so:

`const names = res.data.results.map((name) => { return { ... { name } }})`

The result is the exact same, but without the `for` loop 👍🏼